3D Chess/Assets/Scripts/PieceEntity.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Realms;
+using UnityEngine;
+
+public class PieceEntity : RealmObject
+{
+    public int Type { get; set; }
+
+    public float PositionX { get; set; }
+    public float PositionY { get; set; }
+    public float PositionZ { get; set; }
```

Comment:
These properties can also be private since we're using `GetPosition`/`SetPosition` to manipulate the entity.

Reply:
Actually, I can see we're querying on those, so the getters should be public, but the setters may be private.

Reply:
Yeah, unfortunately we need to query the actual Realm properties directly. Is there a better way to do this without exposing them directly?

If adjusted it with 6d79643 partially but maybe there is an even better solution.

Reply:
Not yet, unfortunately. Cocoa will be spearheading an effort to provide custom type mappings, which will hopefully be done in a way that can also be used in .NET: https://docs.google.com/document/d/16CUK8WjufLb_uuNDDgDuKq4xi60CCZApZARehHoqxdM/edit.

Reply:
The setters can still be private though:

```suggestion
    public float PositionX { get; private set; }
    public float PositionY { get; private set; }
    public float PositionZ { get; private set; }
```

Reply:
Absolutely right. 👍